src/features/device/deviceSlice.ts

```diff
@@ -52,39 +50,35 @@ export const deviceSlice = createSlice({
       if (action.payload) {
         // If whatever is passed in when this is called is not null
         state.activeWaypoint = null;
-        state.showAlgosAccordion = false;
+        state.infoPane = null;
       }
     },
 
-    setActiveWaypoint: (
-      state,
-      action: PayloadAction<Waypoint["id"] | null>
-    ) => {
+    setActiveWaypoint: (state, action: PayloadAction<Waypoint | null>) => {
       state.activeWaypoint = action.payload;
       if (action.payload) {
-        // If whatever is passed in when this is called is not null; want only one active of either node, wp, or accordion
+        // If there is an active waypoint then we don't want another info screen
+        state.infoPane = "waypoint";
         state.activeNode = null;
-        state.showAlgosAccordion = false;
-        // Only want edit to be true if someone explicitly clicks on edit; prevent holdover from exiting previous node exit
-        state.waypointEdit = false;
       }
     },
 
-    setWaypointEdit: (state, action: PayloadAction<boolean>) => {
-      state.waypointEdit = action.payload;
-    },
-
-    setAllowOnMapWaypointCreation: (state, action: PayloadAction<boolean>) => {
-      state.allowOnMapWaypointCreation = action.payload;
-    },
+    setInfoPane: (
+      state,
+      action: PayloadAction<"waypoint" | "waypointEdit" | "algos" | null>
```

Comment:
Sure I'd be down with that, can I ask why?

Reply:
I think the reason I had left this comment was because the type of the active pane can't be accessed by other files without using some funky `typeof` constructs. Looking back on this, my comment was misleading, I think the big issue I was having was just not having the union type not declared outside of the `setInfoPane` method. Nothing particularly special about enums, although I just found this article that you might find interesting ([link](https://www.bam.tech/article/should-you-use-enums-or-union-types-in-typescript))